rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/validation/CachingValidator.java

```diff
@@ -117,7 +119,11 @@ protected boolean validateCachingBehavior()
         fileLastModified, VALIDATOR_START_BLOCK, VALIDATOR_END_BLOCK).setClusterType(VALIDATOR_CLUSTER_TYPE);
 
     try {
-      List<BlockLocation> locations = bookKeeper.getCacheStatus(request);
+      CacheStatusResponse response = bookKeeper.getCacheStatus(request);
+      if (response == null) {
```

Comment:
this can be clubbed into one condition as:
```
if (response == null || response.getBlocks().isEmpty() || response.getBlocks().get(0).getLocation() != Location.LOCAL) {
        return false;	        return false;
}
```